geomstats/geometry/graphspace.py

```diff
@@ -0,0 +1,258 @@
+"""Graph Space."""
+
+import geomstats.backend as gs
+from geomstats.geometry.matrices import Matrices, MatricesMetric
+
+
+class _GraphSpace:
+    r"""Class for the Graph Space.
+
+    Graph Space to analyse populations of labelled and unlabelled graphs.
+    The space is a quotient space obtained by applying the permutation
+    action of nodes to the space of adjacency matrices.
+
+    Points are represented by :math:`nodes \times nodes` adjacency matrices.
+
+    Parameters
+    ----------
+    nodes : int
+        Number of graph nodes
+    p : int
+        Dimension of euclidean parameter or label associated to a graph.
+
+    References
+    ----------
+    ..[Calissano2020]  Calissano, A., Feragen, A., Vantini, S.
+              “Graph Space: Geodesic Principal Components for aPopulation of
+              Network-valued Data.”
+              Mox report 14, 2020.
+              https://mox.polimi.it/reports-and-theses/publication-results/?id=855.
+    """
+
+    def __init__(self, nodes, p=None):
+        self.nodes = nodes
+        self.p = p
+        self.adjmat = Matrices(self.nodes, self.nodes)
+
+    def belongs(self, graph, atol=gs.atol):
+        r"""Check if the matrix is an adjacency matrix associated to the
+        graph with n nodes.
+
+        Parameters
+        ----------
+        graph : array-like, shape=[..., n, n]
+            Matrix to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,n]
+            Boolean denoting if graph belongs to the space.
+        """
+        return self.adjmat.belongs(graph, atol=atol)
+
+    def random_point(self, n_samples=1, bound=1.0):
+        r"""Sample in Graph Space.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+        bound : float
+            Bound of the interval in which to sample in the tangent space.
+            Optional, default: 1.
+
+        Returns
+        -------
+        graph_samples : array-like, shape=[..., n, n]
+            Points sampled in GraphSpace(n).
+        """
+        return self.adjmat.random_point(n_samples=n_samples, bound=bound)
+
+    def permute(self, graph_to_permute, permutation):
+        r"""Permutation action applied to graph observation.
+
+        Parameters
+        ----------
+        graph_to_permute : array-like, shape=[..., n, n]
+            input graphs to be permuted
```

Comment:
NIT: Input graphs to be permuted. 

(upper case at the first letter and "." at the end)